src/main/java/mab/task/Task.java

```diff
@@ -0,0 +1,63 @@
+package mab.task;
+
+/**
+ * Abstract base class representing a task with common attributes.
+ * Provides foundation for task management and persistence.
+ */
+public abstract class Task {
+    String text;
+    boolean done;
+
+    /**
+     * Creates a new generic task.
+     * 
+     * @param t Text description of task
+     * @param d Initial completion status
+     */
+    public Task(String t, boolean d) {
+        text = t; done = d;
```

Comment:
Statements should be on separate lines
```suggestion
        text = t; 
        done = d;
```

Reply:
Thanks for the suggestions!